### ISSUE TITLE
Allow flexible `basePath`

### ISSUE THREAD

**Issue by ax3l** [ComputationalRadiationPhysics/openPMD-standard#15]
In future versions, a flexible `basePath` should be allowed.

For that we have to provide easy-to-user helper functions for both `fileBased` and `groupBased` files.

**Comment by ChristopherMayes** [comment]
I agree. In many of my uses, I would like to set `basePath`  and `particlesPath` to `/` . Could this be done with the 2.0 standard?

**Comment by DavidSagan** [comment]
I agree with Chris. This should be an easy thing to specify and it would save code maintainers time to implement this with 2.0 rather than later.

**Comment by ax3l** [comment]
Sounds good! :+1: 
(I am on vacation from today until next Sunday, but placed it already on the list.)

**Comment by DavidSagan** [comment]
So I propose that basePath can be any absolute (must begin with "/") directory specification. If basePath does not have a "%T" then there is no more than one messes group and no more than one particle group.

Question: Should constructs like "/xy%Tz/" be accepted? This would match to "/xy001z/" etc.

**Comment by tybulut** [comment]
@ax3l , it looks like this issue is not accepted yet. What is the process for this to be "Accepted"?

It will be great to update the description of `basePath`:

allowed value: a string full path that starts with `/` ends with `/%T/`. Recommended value is  `/data/%T/`

And update `iterationFormat` inherit it from:

for `groupBased`:
- Must be equal to and encoded in the `basePath`, for example `/data/%T/`

**Comment by tybulut** [comment]
@ax3l , it looks like this issue is not accepted yet. Could you let us know what the process is for this to be "Accepted"?

It will be great to update the description of `basePath`:

- allowed value: a string full path that starts with `/` ends with `%T/`. Recommended value is  `/data/%T/`

And update `iterationFormat` inherit it from:

for `groupBased`:
- Must be equal to and encoded in the `basePath`, for example `/data/%T/`
